### ISSUE TITLE
SDMMC no longer working for >= 0.12.0

### ISSUE THREAD

**Issue by ostenning** [stm32-rs/stm32h7xx-hal#360]
Upon upgrading to `>= 0.12.0` the `sdmmc` implementation fails with Timeout errors. 
Its odd because the sdcard is detected and I can get the status of it, but when attempting to write data it fails.
I seem to be able to read data fine with `embedded-sdmmc` and fat32, but writing data causes an immediate Device Timeout error.

I downgraded to SHA `8a74bb4e950166550ea4d3d23e8835d48c55b221` and the problems were immediately fixed.

cc. @richardeoin

**Comment by richardeoin** [comment]
@ostenning [The example](https://github.com/stm32-rs/stm32h7xx-hal/blob/master/examples/sdmmc.rs) works fine for me on v0.12. Tested on a STM32H747I-DISCO development board with a SanDisk Extreme 32 GB SDHC UHS-I card.

```
INFO  examples/sdmmc.rs:144 - ----------------------
INFO  examples/sdmmc.rs:145 -
INFO  examples/sdmmc.rs:161 - Read 10 blocks at 24076.043 bytes/s
INFO  examples/sdmmc.rs:162 -
INFO  examples/sdmmc.rs:170 - Wrote block 0
INFO  examples/sdmmc.rs:170 - Wrote block 1
INFO  examples/sdmmc.rs:170 - Wrote block 2
INFO  examples/sdmmc.rs:170 - Wrote block 3
INFO  examples/sdmmc.rs:170 - Wrote block 4
INFO  examples/sdmmc.rs:170 - Wrote block 5
INFO  examples/sdmmc.rs:170 - Wrote block 6
INFO  examples/sdmmc.rs:170 - Wrote block 7
INFO  examples/sdmmc.rs:170 - Wrote block 8
INFO  examples/sdmmc.rs:170 - Wrote block 9
INFO  examples/sdmmc.rs:182 - Done!
```

The recent refactor #341 changed the status check at the end of `write_block` [to use CMD13](https://github.com/stm32-rs/stm32h7xx-hal/blob/master/src/sdmmc.rs#L722). That matches the F4 HAL, which has been well tested. (Previously this HAL used ACMD13). That might be a clue, but unless you have more details or a solution I have to assume the issue is your setup.

**Comment by richardeoin** [comment]
Ok I can reproduce by making multiple writes in succession, without a delay or read in between. There's a fix in #361 , and I also extended the example to cover this behaviour.

**Comment by ostenning** [comment]
Good to hear you were able to reproduce it @richardeoin, thanks for the fix ☺️

**Comment by richardeoin** [comment]
Closed by #361
